### PR TITLE
Update cucumber testsuite module to match new naming

### PR DESCRIPTION
**Pull request by maximenoel8**
## What does this PR change?

Update cucumber testsuite module to match new naming.
New hostname naming:
  -  for suse minions :
     -   suse-minion
     -   suse-sshminion
     -   suse-client
     -   slemicro-minion
  -  for redhat / rocky8 minions:
     -    rhlike-minion
     -    rhlike-sshminion
  -  for debian / ubuntu2204 minions:
     -   deblike-minion
     -   deblike-sshminion

Module names are the same using an `_` instead of `-`

Also change the default value for `install_salt_bundle` from _false_ to _true_ (align with module default value). It will remove the need of adding https://github.com/SUSE/susemanager-ci/blob/8ed2e0bcba38461cf5d0e445d75264a3d1a31a51/terracumber_config/tf_files/SUSEManager-4.3-NUE.tf#L204 for all pipelines using cucumber_testsuite module.

Related to: https://github.com/SUSE/spacewalk/issues/25062

Discussion: https://suse.slack.com/archives/C02D134507M/p1731289062684489

### CODE REVIEW

File: modules/acceptance_tests_module/main.tf [hunk 1]
```diff
@@ -0,0 +1,516 @@
+
```
Comment (srbarrios) [inline modules/acceptance_tests_module/main.tf:1]: Let's please keep the same name we have "cucumber_testsuite"

File: README_TESTING.md [hunk 2]
```diff
@@ -132,23 +132,23 @@ Several test hosts are optional and can be activated via a `host_settings` block
...
+  suse_client = {
   }
-  suse-minion = {
+  suse_minion = {
   }
-  suse-sshminion = {
+  suse_sshminion = {
   }
-  redhat-minion = {
+  redlike_minion = {
```
Comment (srbarrios) [inline README_TESTING.md:141]:
```suggestion
  rhlike_minion = {
```

File: modules/cucumber_testsuite/main.tf [hunk 3]
```diff
@@ -42,7 +42,7 @@ locals {
   names                     = { for host_key in local.hosts :
     host_key => lookup(var.host_settings[host_key], "name", null) if var.host_settings[host_key] != null ? contains(keys(var.host_settings[host_key]), "name") : false }
   install_salt_bundle       = { for host_key in local.hosts :
-    host_key => lookup(var.host_settings[host_key], "install_salt_bundle", false) if var.host_settings[host_key] != null }
+    host_key => lookup(var.host_settings[host_key], "install_salt_bundle", true) if var.host_settings[host_key] != null }
```
Comment (srbarrios) [inline modules/cucumber_testsuite/main.tf:45]: Fine with this, not sure if that should be in another PR.
Anyway, let's at least comment it on the description of the PR please.

File: modules/cucumber_testsuite/main.tf [hunk 4]
```diff
@@ -232,252 +232,253 @@ module "proxy_containerized" {
...
-  kvmhost_configuration          = contains(local.hosts, "kvm-host") ? module.kvm-host.configuration : { hostnames = [], ids = [], ipaddrs = [], macaddrs = [], private_macs = [] }
-  monitoringserver_configuration = contains(local.hosts, "monitoring-server") ? module.monitoring-server.configuration : { hostnames = [], ids = [], ipaddrs = [], macaddrs = [], private_macs = [] }
+  client_configuration           = contains(local.hosts, "suse_client") ? module.suse_client.configuration : { hostnames = [], ids = [], ipaddrs = [], macaddrs = [], private_macs = [] }
+  minion_configuration           = contains(local.hosts, "suse_minion") ? module.suse_minion.configuration : { hostnames = [], ids = [], ipaddrs = [], macaddrs = [], private_macs = [] }
+  sshminion_configuration        = contains(local.hosts, "suse_sshminion") ? module.suse_sshminion.configuration : { hostnames = [], ids = [], ipaddrs = [], macaddrs = [], private_macs = [] }
+  slemicro_minion_configuration  = contains(local.hosts, "slemicro_minion") ? module.slemicro_minion.configuration : { hostnames = [], ids = [], ipaddrs = [], macaddrs = [], private_macs = [] }
```
Comment (srbarrios) [inline modules/cucumber_testsuite/main.tf:474]: I wonder if for those we want also to rename the configuration variable name, to have something like... `suse_minion_configuration` for example.

**Review by srbarrios — APPROVED** [review]

File: modules/cucumber_testsuite/main.tf (hunk 4 above)
Comment (maximenoel8) [reply modules/cucumber_testsuite/main.tf:474]: Definitely. I added a todo for that. I was looking at the redhat, debian conf but those confs too. I will need those changes in a separate PR because it also require some changes on controller salt states.

**Review by NamelessOne91 — APPROVED** [review]
I haven't spotted any more typos or the likes before my eye started crossing 😄 

Thanks Maxime.
I think we' may need to update the 4.3 BV and CI feature for the containerized proxy once the host-name change takes place.

**Comment by maximenoel8** [comment]
> I haven't spotted any more typos or the likes before my eye started crossing 😄
> 
> Thanks Maxime. I think we' may need to update the 4.3 BV and CI feature for the containerized proxy once the host-name change takes place.

I'm not sure what need to be changes. Don't hesitate to create an issue and assign it to me we the changes you see.
